pkg/build/graph/edges.go

```diff
@@ -56,7 +57,8 @@ func AddInputOutputEdges(g osgraph.MutableUniqueGraph, node *buildgraph.BuildCon
 			in := imagegraph.FindOrCreateSyntheticImageStreamTagNode(g, imagegraph.MakeImageStreamTagObjectMeta(defaultNamespace(from.Namespace, node.BuildConfig.Namespace), name, tag))
 			g.AddEdge(in, node, BuildInputImageEdgeKind)
 		case "ImageStreamImage":
-			glog.V(4).Infof("Ignoring ImageStreamImage reference in BuildConfig %s/%s", node.BuildConfig.Namespace, node.BuildConfig.Name)
+			in := imagegraph.EnsureImageStreamImageNode(g, defaultNamespace(from.Namespace, node.BuildConfig.Namespace), from.Name)
```

Comment:
Is there a very compelling reason not to follow the flow of 
 1. create all nodes
 1. create all edges
 1. if a node is not present, create a synthetic node to indicate it is missing?

Looks like `EnsureDockerRepositoryNode` in this method slipped through, but I've been trying to squash all of them.  Creating random nodes while adding edges makes it impossible to determine when the graph is incomplete and something is wrong in the project.